juju/juju
# Use whitelist when adding clouds.

## Description of change

In multi-cloud Juju world, it is easy to add clouds to a bootstrapped controller. However, some clouds are less compatible with each other. This PR introduces a whitelist of the clouds that Juju can support easily out of the box.

With all other cloud combinations, users need to be absolutely sure that they know what they are doing (networks and connectivity needs to be configured outside of Juju) and can use --force option to force cloud addition.

'add-cloud' help also provides whitelist now - https://pastebin.ubuntu.com/p/WXc9rnTv5J/

'add-model' already prevents users from adding models on a clouds that are not yet known to the controller (https://pastebin.ubuntu.com/p/5h496RkqYw/) so there is no need to provide a --force option here.

As a drive-by, this PR renames some variables that have been conflicting with imports names.

## QA steps
1. bootstrap a controller on a non-whitelisted cloud and enable multi-cloud feature for it. I used aws
2. Try adding cloud. You should be stopped -  https://pastebin.ubuntu.com/p/Gc6KRZPp3Z/
3. Using --force allows operation to proceed
4. Bootstrap a controller on a whitelisted cloud and enable multi-cloud feature for it. I used maas
5. Try adding non-whitelisted cloud. You should be stopped. I used lxd

## Bug reference

https://bugs.launchpad.net/juju/+bug/1825889
